src/analysis/retail/evoker/preservation/normalizers/CastLinkNormalizer.ts

```diff
@@ -329,6 +329,16 @@ const EVENT_LINKS: EventLink[] = [
       );
     },
   },
+  // link dream breath hot application to cast
+  {
+    linkRelation: FROM_HARDCAST,
+    linkingEventId: [SPELLS.DREAM_BREATH.id],
+    linkingEventType: [EventType.ApplyBuff],
+    referencedEventId: [TALENTS_EVOKER.DREAM_BREATH_TALENT.id],
+    referencedEventType: [EventType.Cast],
+    backwardBufferMs: 2500, // there is some pretty big latency in logs, so being generous
```

Comment:
This is not latency, the cast event is timestamped at the start of the empower. ApplyBuff & RemoveBuff events are used to track the length of the empowered cast so it might make more sense to set up links for the empowered cast time to link this up. 
![image](https://user-images.githubusercontent.com/26779541/206205418-a274de98-e34a-4c9e-81b0-755aa2d5ad3b.png)
Or you could just group the buff applied events like you've done with TA/EB to track targets hit because dream breath echo is its own spellid so you don't have to worry about collision there.
The only concern  with that would be stasis potentially

Reply:
This appears to be how it works with fire breath as well, but I am not seeing it work with this way for spiritbloom